solc/CommandLineInterface.cpp

```diff
@@ -509,7 +511,32 @@ Allowed options)",
 
 bool CommandLineInterface::processInput()
 {
-	readInputFilesAndConfigureRemappings();
+	// determine if we should use stdin
+	bool usestdin = false;
+	vector<string> inputFiles;
+
+	if (!m_args.count("input-file")) {
+		inputFiles = vector<string>();
+		usestdin = true;
+	} else {
+		inputFiles = m_args["input-file"].as<vector<string>>();
+
+		if (inputFiles.size() == 0)
+			usestdin = true;
+		else
+			// search for "-" in input files to enable stdin (and filter out the "-")
+			for (unsigned idx = 0; idx < inputFiles.size(); ++idx) {
+				string path = inputFiles[idx];
+				if (path == "-") {
+					inputFiles.erase(inputFiles.begin() + idx);
```

Comment:
I think it would be better to loop through `m_args["input-fil"].as<vector<string>>()` and add everything to `inputFiles` that is not `"-"`.

Also the current routine does not work correctly if you specify `-` twice.